common/test/src/ui/tst_MapDocument.cpp

```diff
@@ -462,6 +463,76 @@ TEST_CASE_METHOD(MapDocumentTest, "canUpdateLinkedGroups")
     document->canUpdateLinkedGroups(kdl::vec_static_cast<mdl::Node*>(entityNodes)));
 }
 
+TEST_CASE_METHOD(MapDocumentTest, "newDocument")
+{
+  const auto fgdContent = R"(
+    @SolidClass = worldspawn : "World entity" [
+      some_default_prop(string) : "" : "value"
+    ]
+  )";
+
+  const auto configTemplate = R"(
+    {
+        version: 9,
+        name: "testNewDocument",
+        icon: "Icon.png",
+        "fileformats": [ { "format": "Standard" } ],
+        "filesystem": {
+            "searchpath": ".",
+            "packageformat": { "extension": ".pak", "format": "idpak" }
+        },
+        "materials": {
+            "root": "textures",
+            "format": { "extensions": [".jpg"], "format": "image" }
+        },
+        "entities": {
+            "definitions": [ "WorldSpawnOnly.fgd" ],
+            "defaultcolor": "0.6 0.6 0.6 1.0",
+            "setDefaultProperties": PLACEHOLDER
+        },
+        "tags": { "brush": [], "brushface": [] },
+        "faceattribs": { "surfaceflags": [], "contentflags": [] }
+    }
+  )";
+
+  SECTION("Default worldspawn properties")
+  {
+    using T = std::tuple<bool, std::vector<mdl::EntityProperty>>;
+    const auto& [setDefaultProperties, expectedEntityProperties] = GENERATE(values<T>({
+      {false,
+       {{mdl::EntityPropertyKeys::Classname,
+         mdl::EntityPropertyValues::WorldspawnClassname}}},
+      {true,
+       {{mdl::EntityPropertyKeys::Classname,
+         mdl::EntityPropertyValues::WorldspawnClassname},
+        {"some_default_prop", "value"}}},
+    }));
+
+    std::string configContent = configTemplate;
```

Comment:
Please use auto.

Reply:
Please use `fmt::format` with placeholders instead of replace. You will have to escape every opening / closing brace `{` by repeating it: `{{`, then you can use `{}` for the placeholders.

Reply:
So it would look like `const auto configContent = fmt::format(configTemplate, setDefaultProperties ? "true" : "false");`

Reply:
Sorry, but I couldn't use `auto` because its becomes a `const char*`, and what I need is a `std::string` to use replace methods.
I can change it to use `fmt::format` but I think it would make the raw string harder to read with all the brackets being escaped and error prone if you miss one escape. Still, I can make it that way if you want it.

Reply:
Please use fmt, I'd like to use that for string formatting wherever possible now.

Reply:
Done, I had to use fmt::runtime, otherwise it won't compile.